core-web/libs/block-editor/src/lib/shared/services/ai-content/ai-content.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Observable, throwError } from 'rxjs';
+
+import { DOCUMENT } from '@angular/common';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable, Inject } from '@angular/core';
+
+import { catchError, map } from 'rxjs/operators';
+
+interface OpenAIResponse {
+    model: string;
+    prompt: string;
+    response: string;
+}
+
+@Injectable()
+export class AiContentService {
+    constructor(private http: HttpClient, @Inject(DOCUMENT) private document: Document) {}
+
+    get window(): Window {
+        return this.document.defaultView || window;
+    }
+
+    getIAContent(prompt: string): Observable<string> {
+        const url = `${this.window.origin}/api/ai/text/generate`;
```

Comment:
I'm not sure, but I think it should work with just `/api/ai/text/generate`

Reply:
You are right